models/points.go

```diff
@@ -2100,6 +2161,34 @@ func (a Tags) Clone() Tags {
 	return others
 }
 
+// KeyValues returns the Tags as a list of key, value pairs,
+// maintaining the original order of a. v will be used if it has
+// capacity.
+func (a Tags) KeyValues(v [][]byte) [][]byte {
+	l := a.Len() * 2
+	if cap(v) < l {
+		v = make([][]byte, 0, l)
+	} else {
+		v = v[:l]
+	}
+	for i := range a {
+		v = append(v, a[i].Key, a[i].Value)
+	}
+	return v
+}
+
+// sorted returns true if a is sorted and is an optimization
+// to avoid an allocation when calling sort.IsSorted, improving
+// performance as much as 50%.
+func (a Tags) sorted() bool {
+	for i := len(a) - 1; i > 0; i-- {
+		if string(a[i].Key) < string(a[i-1].Key) {
+			return false
+		}
```

Comment:
Does `string(a[i].Key)` do an allocation? Would `bytes.Compare()` be faster?

Reply:
It won't allocate, however, I will change the code to be clearer per your suggestion.

Incidentally, I reported that `bytes.Equal` was also slower than `string(a) == string(b)` in the Gopher `#performance` slack and someone logged an issue. Go 1.13 was updated to use the cast for `bytes.Equal`:

https://github.com/golang/go/blob/master/src/bytes/bytes.go#L20

Reply:
Done